app/models/partner.rb

```diff
@@ -91,10 +91,16 @@ class Partner < ApplicationRecord
 
   scope :for_site, lambda { |site|
     site_neighbourhood_ids = site.neighbourhoods.map(&:subtree).flatten.map(&:id)
+    site_tag_ids = site.tags.map(&:id)
 
-    joins('left join addresses on addresses.id = partners.address_id')
-      .joins('left join service_areas on partners.id = service_areas.partner_id')
-      .where('(service_areas.neighbourhood_id in (?)) or (addresses.neighbourhood_id in (?))', site_neighbourhood_ids, site_neighbourhood_ids)
+    partners = joins('left join addresses on addresses.id = partners.address_id')
+               .joins('left join service_areas on partners.id = service_areas.partner_id')
+               .where('(service_areas.neighbourhood_id in (?)) or (addresses.neighbourhood_id in (?))',
+                      site_neighbourhood_ids, site_neighbourhood_ids)
+
+    return partners.of_tag(site_tag_ids) unless site_tag_ids.empty?
```

Comment:
feel like this is a bit of an obfuscated if else currently maybe just refactor as that?